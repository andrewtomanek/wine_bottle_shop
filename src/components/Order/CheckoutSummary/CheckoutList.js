import React from "react";

import classes from "./CheckoutList.module.css";
import CheckoutItem from "./CheckoutItem";

const store = props => {
  let transformedItems = Object.keys(props.inventory)
    .map(igKey => {
      return [...Array(props.inventory[igKey])].map((_, i) => {
        return (
          <CheckoutItem
            key={igKey + i}
            type={igKey}
            itemName={props.listItems[igKey]}
          />
        );
      });
    })
    .reduce((arr, el) => {
      return arr.concat(el);
    }, []);

  if (transformedItems.length === 0) {
    transformedItems = <p>Naplň přepravní box</p>;
  }
  return <div className={classes.CheckoutList}>{transformedItems}</div>;
};

export default store;
