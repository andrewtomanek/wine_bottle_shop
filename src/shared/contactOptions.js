export const validationData = {
  name: {
    elementType: "input",
    elementConfig: {
      type: "text",
      placeholder: "Jméno a Příjmení",
    },
    value: "",
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
  },
  street: {
    elementType: "input",
    elementConfig: {
      type: "text",
      placeholder: "Ulice, č.p.",
    },
    value: "",
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
  },
  zipCode: {
    elementType: "input",
    elementConfig: {
      type: "text",
      placeholder: "PSČ",
    },
    value: "",
    validation: {
      required: true,
      minLength: 5,
      maxLength: 5,
      isNumeric: true,
    },
    valid: false,
    touched: false,
  },
  country: {
    elementType: "input",
    elementConfig: {
      type: "text",
      placeholder: "Město",
    },
    value: "",
    validation: {
      required: true,
    },
    valid: false,
    touched: false,
  },
  email: {
    elementType: "input",
    elementConfig: {
      type: "email",
      placeholder: "E-Mail",
    },
    value: "",
    validation: {
      required: true,
      isEmail: true,
    },
    valid: false,
    touched: false,
  },
  deliveryMethod: {
    elementType: "select",
    elementConfig: {
      options: [
        { value: "cash", displayValue: "Platba  Dobírkou" },
        { value: "card", displayValue: "Platba Hotově/kartou" },
        { value: "online", displayValue: "Online platba kartou" },
      ],
    },
    value: "online",
    validation: {},
    valid: true,
  },
};
