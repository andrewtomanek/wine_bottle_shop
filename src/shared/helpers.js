export const reduceCartState = (inventory) => {
  const sum = Object.keys(inventory)
    .map((igKey) => {
      return inventory[igKey];
    })
    .reduce((sum, el) => {
      return sum + el;
    }, 0);
  return sum;
};
