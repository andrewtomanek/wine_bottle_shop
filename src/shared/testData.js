export const inventory = {
  shopItem1: 1,
  shopItem2: 2,
  shopItem3: 3,
  shopItem4: 4,
  shopItem5: 5,
  shopItem6: 6,
  shopItem7: 7,
  shopItem8: 8,
};

export const listItems = {
  shopItem1: "Solaris",
  shopItem2: "Dornfelder",
  shopItem3: "Agni",
  shopItem4: "Muscaris",
  shopItem5: "Sauvignon",
  shopItem6: "Tramín",
  shopItem7: "Hibernal",
  shopItem8: "Pálava",
};

export const pricesList = {
  shopItem1: 50,
  shopItem2: 100,
  shopItem3: 150,
  shopItem4: 70,
  shopItem5: 140,
  shopItem6: 90,
  shopItem7: 120,
  shopItem8: 110,
};
