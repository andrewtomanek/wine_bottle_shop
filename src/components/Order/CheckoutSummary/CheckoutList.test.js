import React from "react";
import { shallow } from "enzyme";
import CheckoutList from "./CheckoutList";
import { inventory, listItems, pricesList } from "../../../shared/testData";

const props = {
  inventory,
  listItems,
  pricesList,
};

const emptyProps = { inventory: {} };

describe("Shallow Render CheckoutList component", () => {
  it("renders Complete", () => {
    const wrapper = shallow(<CheckoutList {...props} />);
    expect(wrapper.find(".PriceSum").text()).toContain(" Komplet ");
  });

  it("renders empty cart", () => {
    const wrapper = shallow(<CheckoutList {...emptyProps} />);
    expect(wrapper.find(".Empty").text()).toContain("Žádné položky");
  });

  it("renders shop items", () => {
    const wrapper = shallow(<CheckoutList {...props} />);
    expect(wrapper.find(".CheckoutList").children()).toHaveLength(36);
  });

  it("Match Snapshot of CheckoutList component", () => {
    const wrapper = shallow(<CheckoutList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
