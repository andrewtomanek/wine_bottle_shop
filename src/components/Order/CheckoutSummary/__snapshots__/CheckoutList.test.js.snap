// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render CheckoutList component Match Snapshot of CheckoutList component 1`] = `
<Fragment>
  <div
    className="CheckoutList"
  >
    <Connect(CheckoutCard)
      itemName="Solaris"
      itemPrice={50}
      itemType="shopItem1"
      key="shopItem10"
    />
    <Connect(CheckoutCard)
      itemName="Dornfelder"
      itemPrice={100}
      itemType="shopItem2"
      key="shopItem20"
    />
    <Connect(CheckoutCard)
      itemName="Dornfelder"
      itemPrice={100}
      itemType="shopItem2"
      key="shopItem21"
    />
    <Connect(CheckoutCard)
      itemName="Agni"
      itemPrice={150}
      itemType="shopItem3"
      key="shopItem30"
    />
    <Connect(CheckoutCard)
      itemName="Agni"
      itemPrice={150}
      itemType="shopItem3"
      key="shopItem31"
    />
    <Connect(CheckoutCard)
      itemName="Agni"
      itemPrice={150}
      itemType="shopItem3"
      key="shopItem32"
    />
    <Connect(CheckoutCard)
      itemName="Muscaris"
      itemPrice={70}
      itemType="shopItem4"
      key="shopItem40"
    />
    <Connect(CheckoutCard)
      itemName="Muscaris"
      itemPrice={70}
      itemType="shopItem4"
      key="shopItem41"
    />
    <Connect(CheckoutCard)
      itemName="Muscaris"
      itemPrice={70}
      itemType="shopItem4"
      key="shopItem42"
    />
    <Connect(CheckoutCard)
      itemName="Muscaris"
      itemPrice={70}
      itemType="shopItem4"
      key="shopItem43"
    />
    <Connect(CheckoutCard)
      itemName="Sauvignon"
      itemPrice={140}
      itemType="shopItem5"
      key="shopItem50"
    />
    <Connect(CheckoutCard)
      itemName="Sauvignon"
      itemPrice={140}
      itemType="shopItem5"
      key="shopItem51"
    />
    <Connect(CheckoutCard)
      itemName="Sauvignon"
      itemPrice={140}
      itemType="shopItem5"
      key="shopItem52"
    />
    <Connect(CheckoutCard)
      itemName="Sauvignon"
      itemPrice={140}
      itemType="shopItem5"
      key="shopItem53"
    />
    <Connect(CheckoutCard)
      itemName="Sauvignon"
      itemPrice={140}
      itemType="shopItem5"
      key="shopItem54"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem60"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem61"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem62"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem63"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem64"
    />
    <Connect(CheckoutCard)
      itemName="Tramín"
      itemPrice={90}
      itemType="shopItem6"
      key="shopItem65"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem70"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem71"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem72"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem73"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem74"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem75"
    />
    <Connect(CheckoutCard)
      itemName="Hibernal"
      itemPrice={120}
      itemType="shopItem7"
      key="shopItem76"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem80"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem81"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem82"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem83"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem84"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem85"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem86"
    />
    <Connect(CheckoutCard)
      itemName="Pálava"
      itemPrice={110}
      itemType="shopItem8"
      key="shopItem87"
    />
  </div>
  <p
    className="PriceSum"
  >
     Komplet 
  </p>
</Fragment>
`;
